nav2_controller/src/nav2_controller.cpp

```diff
@@ -169,7 +169,7 @@ void ControllerServer::followPath()
 
       if (action_server_->is_cancel_requested()) {
         RCLCPP_INFO(get_logger(), "Goal was canceled. Stopping the robot.");
-        action_server_->terminate_goals();
```

Comment:
So these are terminate all’s. I think the failure cases for a specific goal (unable to compute path, exception, etc) should terminate current, and the only terminate all’s should be when an external user says for the server to cancel. 

What do you think?

Reply:
Makes sense to terminate only the current goal for basic failure cases.

Other runtime exceptions might require to terminate both before getting into more aggressive recovery.

Reply:
I'll replace the terminate all's with only current. The action server now checks the status of any pending goals and executes or exits.